### PR TITLE
Home task 4: Implemented “Add movie”, “Edit”, “Delete” components for…

### PR DESCRIPTION
**Pull request by HodynskyiYK**
… home page and configured linter with prettier

### CODE REVIEW

File: src/components/modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        <div className={styles.modal}>
+            <div className={styles.modalContent}>
+                <button
+                    className={styles.closeModal}
+                    onClick={hideModal}
+                >
+                    <svg width="23" height="23" viewBox="0 0 23 23" fill="none" xmlns="http://www.w3.org/2000/svg">
+                        <path d="M1.47099 1.15426L21.529 21.2122" stroke="white" strokeWidth="2" strokeLinecap="round"/>
+                        <path d="M21.529 1.15426L1.47103 21.2122" stroke="white" strokeWidth="2" strokeLinecap="round"/>
+                    </svg>
```
Comment (oleksapro) [inline src/components/modal/Modal.tsx]: Better keep svg in assets

File: src/components/button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React, { FC } from 'react'
+import classnames from 'classnames'
+import { IButton } from './buttonTypes'
+import styles from './Button.module.scss'
+
+export const Button: FC<IButton> = ({buttonClassName, buttonAction, buttonText}) => {
+
+    return (
+        <button
+            className={classnames(styles.btn, `${styles[buttonClassName]}`)}
```
Comment (oleksapro) [inline src/components/button/Button.tsx:10]: It is up to you, but I'd rather use abbreviation for classnames, `cn`

File: src/pages/Home/components/delete-movie-modal/DeleteMovieModal.module.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
+@import "../../../../styles/stylesVariables";
```
Comment (oleksapro) [inline src/pages/Home/components/delete-movie-modal/DeleteMovieModal.module.scss]: You can use absolute imports `import "src/styles/stylesVariables";`

File: src/pages/Home/components/delete-movie-modal/DeleteMovieModal.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React, { FC } from 'react'
+import classnames from 'classnames'
+import { Modal } from '../../../../components/modal'
```
Comment (oleksapro) [inline src/pages/Home/components/delete-movie-modal/DeleteMovieModal.tsx:3]: You can use absolute import with some configuration https://create-react-app.dev/docs/importing-a-component/#absolute-imports

File: src/pages/Home/components/movie-card/MovieCard.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                <div className={styles.cardHeader}>
+                    <a
+                        href={movieItem.link}
+                        title={movieItem.title}
+                        className={styles.movieLink}
+                    >
+                        <img
+                            className={styles.movieImage}
+                            src={require(`../../../../assets/images/movie-images/${movieItem.image}.png`)}
+                            alt={movieItem.title}
```
Comment (oleksapro) [inline src/pages/Home/components/movie-card/MovieCard.tsx:25]: Better keep all imports on top of file.

File: src/pages/Home/components/movie-card/MovieCard.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                                            <feOffset/>
+                                            <feGaussianBlur stdDeviation="2"/>
+                                            <feColorMatrix type="matrix"
+                                                           values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.196596 0"/>
+                                            <feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow_0_213"/>
+                                            <feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow_0_213"
+                                                     result="shape"/>
+                                        </filter>
+                                    </defs>
+                                </svg>
```
Comment (oleksapro) [inline src/pages/Home/components/movie-card/MovieCard.tsx]: Keep jsx as clean as possible, move svg into assets

File: src/pages/Home/components/movies-list/MoviesList.module.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
+@import "../../../../styles/stylesVariables";
```
Comment (oleksapro) [inline src/pages/Home/components/movies-list/MoviesList.module.scss]: I'd recommend stick to one approach in folders/files naming, if it is folder with component name it in PascalCase.

**Review by oleksapro — COMMENTED** [review]
According to requirements we need to use some lifecycle methods or useEffect. Please use them even if it seems meaningless 😅

File: src/components/modal/Modal.tsx (hunk 1 above)
Comment (HodynskyiYK) [reply src/components/modal/Modal.tsx]: done

File: src/components/button/Button.tsx (hunk 2 above)
Comment (HodynskyiYK) [reply src/components/button/Button.tsx:10]: for me better to keep modules names without abbreviation

File: src/pages/Home/components/delete-movie-modal/DeleteMovieModal.module.scss (hunk 3 above)
Comment (HodynskyiYK) [reply src/pages/Home/components/delete-movie-modal/DeleteMovieModal.module.scss]: For me more understandable use absolute imports only for modules from node_modules

File: src/pages/Home/components/delete-movie-modal/DeleteMovieModal.tsx (hunk 4 above)
Comment (HodynskyiYK) [reply src/pages/Home/components/delete-movie-modal/DeleteMovieModal.tsx:3]: For me more understandable use absolute imports only for modules from node_modules

File: src/pages/Home/components/movie-card/MovieCard.tsx (hunk 5 above)
Comment (HodynskyiYK) [reply src/pages/Home/components/movie-card/MovieCard.tsx:25]: done

File: src/pages/Home/components/movie-card/MovieCard.tsx (hunk 6 above)
Comment (HodynskyiYK) [reply src/pages/Home/components/movie-card/MovieCard.tsx]: done

File: src/pages/Home/components/movies-list/MoviesList.module.scss (hunk 7 above)
Comment (HodynskyiYK) [reply src/pages/Home/components/movies-list/MoviesList.module.scss]: done

**Comment by HodynskyiYK** [comment]
About useEffect: I added it in "src/pages/Home/components/main-content/MainContent.tsx"

**Review by oleksapro — APPROVED** [review]

**Pull request merged**